commander/src/utils/reader.ts

```diff
@@ -236,14 +236,25 @@ const getNestedPropertyTemplate = (schema: Schema): NestedPropertyTemplate => {
 };
 
 const castValue = (
-	strVal: string,
+	val: string,
 	schemaType: string,
-): number | string | Record<string, unknown> => {
+): number | bigint | string | string[] | Record<string, unknown> => {
 	if (schemaType === 'object') {
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
-		return JSON.parse(strVal);
+		return JSON.parse(val);
 	}
-	return Number.isInteger(Number(strVal)) ? Number(strVal) : strVal;
+	if (schemaType === 'array') {
+		return val.split(',');
+	}
+	if (
+		schemaType === 'uint64' ||
+		schemaType === 'sint64' ||
+		schemaType === 'uint32' ||
+		schemaType === 'sint32'
+	) {
+		return schemaType === 'uint64' ? BigInt(val) : Number(val);
+	}
+	return val;
```

Comment:
This missed the condition for `sint64` type. Would be nice to split this condition to two for for bigint and number.